Python/codegen.c

```diff
@@ -795,15 +795,12 @@ codegen_process_deferred_annotations(compiler *c, location loc)
         return SUCCESS;
     }
 
-    instr_sequence *old_instr_seq = INSTR_SEQUENCE(c);
-    instr_sequence *nested_instr_seq = NULL;
     int scope_type = SCOPE_TYPE(c);
-    if (scope_type == COMPILE_SCOPE_MODULE) {
-        nested_instr_seq = (instr_sequence *)_PyInstructionSequence_New();
-        if (nested_instr_seq == NULL) {
+    bool need_separate_block = scope_type == COMPILE_SCOPE_MODULE;
+    if (need_separate_block) {
+        if (_PyCompile_StartAnnotationSetup(c) == ERROR) {
```

Comment:
Great.

We could go one step further:  Call these functions `_PyCompile_EnterAnnotationsSubscope`/`_PyCompile_ExitAnnotationsSubscope` and let the compiler check whether we are in `COMPILE_SCOPE_MODULE` (so need to stash) or not.  Up to you.

Reply:
I'll likely drop the module condition soon for another (more minor) issue, so I'm going to skip that for now. Thanks!